### DISCUSSION TITLE
Ring stream permanently active.

### DISCUSSION THREAD

**Discussion by smith-l** [tsightler/ring-mqtt#664] [Support Q&A] [answered]
Hey - apologies if this is a simple question, but I have a problem where as soon as I add my camera configuration into go2rtc the Ring stream is instantly turned on. I thought it was because a client was connected so I have removed all cameras and just have the camera configured in go2rtc, but still have the problem. I have added a password to ensure it go2rtc that's accessing the stream and it is. My question is have I got the correct live stream link and is it set correctly to only stream on demand?

This is the config in go2rtc - when viewing the stream it connects perfectly., so just want to make sure i'm setting the correct link.

streams:

  camera_whoop:
    - rtsp://lee:********!@03cabcc9-ring-mqtt:8554/5051a961296f_live

Any advise would be great!

thanks 🙏

**Comment by tsightler** [comment]
It's a little difficult to judge something like this without full logs, but it looks like the correct format overall.

**Reply by smith-l** [reply]
That's great. Thank you for confirming. I will investigate further on the client side.

**Reply by tsightler** [reply]
OK, well, if you think you have a case where it is definitely triggering with no external clients able to connect, feel free to open an issue and provide full logs and we can dig into it.  But also note most common issues are things like not setting the still image URL or using features like preload stream in HA.

**Reply by smith-l** [reply]
It's definitely client side as after removing all cameras followed by a restart it's not longer accessing the stream. I need to work out what's going on client side as I have an image url  and the stream set, but it still seems to then access the feed constantly. Thanks for your advise! It's much appreciated.

**Reply by TheFern2** [reply]
I don't know for sure if is a client side issue, restarting HA could have just close the connection.

I saw the same issue with the generic camera in HA. For whatever reason it never closes the stream so the live stream toggle keeps turning on. I simply disabled the camera, and restarted the homeassistant container and that severed the stream connection and the toggle went back to false. This might be a bug, or the generic camera isn't designed to stop streams when closed. I don't know who would be at fault ring-mqtt or the generic camera. I will be happy to provide logs, if you let me know what you need.

Here I tried to turn it off, and it turned on, on its own.
![image](https://github.com/tsightler/ring-mqtt/assets/10265682/6825cb17-f670-4177-8f33-0db346ca6395)

**Reply by tsightler** [reply]
From the perspective of ring-mqtt Home Assistant is the client, so that is still a client side issue.  However, it is well known that HAs native stream component keeps the stream active for 5 minutes after any activity, that is it's designed behavior.  Turning off the stream in ring-mqtt will not stop Home Assistant from attempting to reconnect.  The only valid use case for using the live stream switch to stop the stream is if you started the stream with the switch, for example, an automation that triggers a cloud based recording, that is it.  In any case where the stream is started on-demand trying to stop the stream with the switch will not work.  Well, technically it will stop, but most likely the client will just reconnect.

**Reply by AlbinoSlug** [reply]
This is identical to an issue I faced, which I found to be due to the 'RTSPtoWeb - WebRTC' add on default behaviour. If you're using this also, open the 'RTSPtoWeb - WebRTC' Web UI and edit the Ring stream, then set Stream Type to 'On demand only' (defaults to 'Persistent connection').

Best of luck :)

**Reply by tsightler** [reply]
Thanks for sharing.  This is one of the reasons why the recommended setup is to use the go2rtc addon, which defaults to on-demand, but also has better support for audio codecs, etc.  It should be a drop-in replacement for the RTSPtoWeb/RTSPtoWebRTC addon and still requires the RTSPtoWebRTC integration, and generally requires zero additional configuration.

**Comment by bkr1969** [comment]
OK.  I have completely disabled RTSPto WEB and my doorbell is working, but the stream will not stay off.  Where can I check the persistent/on-demand options for go2rtc?

**Comment by bkr1969** [comment]
I'm adding this in hopes of helping anyone as frustrated as I've been.  First of all, the MQTT_Ring add on is great...thanks for all your work on it.  I removed RTSPtoWEB and WEBRTCCamera integrations and add-ons.  Seems they weren't necessary to get my Eufy cameras working in the first place.  What I think I've figured out is that the persistent stream only occurs when go2rtc is actually open (web UI?)  as soon as I closed this, the persistence stopped.  Now I just have to update Core and hope everything keeps working.  LOL

**Comment by bkr1969** [comment]
Nope.  That's not the answer.  The streaming switch continues to turn on by itself.  HELP!

**Reply by stefan190977** [reply]
gibt es schon eine lösung für das Problem ?

**Comment by danielbelchior** [comment]
Hey, I have the same issue. I am using the RTSP from a generic camera (ring-mqtt) and when I open the stream the connection never close and my ring doorbell stuck in ringing mode.

**Comment by airzl** [comment]
Since the weekend the same issue here. But only on two cameras which have one common denominator that I equipped them with a solar panel to charge the ring battery. Completely confused

**Comment by tsightler** [comment, marked as answer]
I am going to close this discussion as it is not useful.  In the end, there are only three answers:

1) If you restart the addon (not restarting HA or go2rtc, or something else, but restarting the ring-mqtt addon/container itself), and the stream still starts automatically, then some device is requesting a stream and it will not stop until you figure out what device is doing that.  Common issues are things like not setting the image URL correctly, or having no password and some device is finding the unprotected URL and starting the stream.  This is always a good troubleshooting step anyway, changing the livestream username/password and restarting the addon, since that will keep all external clients from connecting.

2) If you restart the addon and the issue stops, then mostly likely it is due to a failure in go2rtc's consumer tracking which can be seen in certain cases where, for example, a client requests a stream but the stream fails to start for some reason (usually a timeout).  There are a few corner cases where, in this situation, go2rtc will get "stuck" thinking that there is a consumer even though there is none, and will keep restarting the stream, however, a simple restart of the addon is all that is required to resolve this.  I've spent a lot of time tracking this issue and I'm hopeful that the next version will reduce the frequency of this issue. 

Also note that the native Home Assistant live live, without go2rtc/WebRTC, will always keep the stream alive for 5 minutes after the last viewer stops, there's nothing that can be done about it.  If you disconnect, wait 4 minutes, then view the stream for 1 second, the 5 minute timer will restart.  Even though this has been documented in the FAQ for ages, some people still seem unaware of this.
